src/classes/RD2_PauseForm_CTRL.cls

```diff
@@ -0,0 +1,287 @@
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Recurring Donations
+* @description Saves and retrieves pause details on a Recurring Donation record
+*/
+public with sharing class RD2_PauseForm_CTRL {
+
+    /***
+    * @description Recurring Donation Schedule service
+    */
+    @TestVisible
+    private static RD2_ScheduleService.PauseScheduleHandler pauseHandler {
+        get {
+            if (pauseHandler == null) {
+                pauseHandler = new RD2_ScheduleService.PauseScheduleHandler();
+            }
+            return pauseHandler;
+        } set;
+    }
+
+    /**
+     * @description Returns PauseData to the LWC so the page can be initialized and rendered.
+     * Determines if the user has the access to create/update the pause
+     * as well as determines if the Recurring Donation is closed.
+     * The Paused Reason field label and options are returned regardless
+     * if the pause exists. However, the schedule Id and the Paused Reason value
+     * are specified if the pause exists for the RD.
+     * @param rdId Recurring Donation Id
+     * @return String JSON string representing PauseData
+     */
+    @AuraEnabled(cacheable=false)
+    public static String getPauseData(Id rdId) {
+        String jsonData;
+
+        try {
+            PauseData pause = new PauseData();
+            pause.rdId = rdId;
+            pause.hasAccess = hasAccess();
+            pause.isRDClosed = pause.hasAccess ? isClosed(rdId) : null;
+
+            if (pause.isRDClosed == false) {
+                pause.pausedReason = buildPausedReason();
+                populateScheduleData(pause, rdId);
+            }
+
+            jsonData = JSON.serialize(pause);
+
+        } catch (Exception e) {
+            throwAuraHandledException(e.getMessage());
+        }
+
+        return jsonData;
+    }
+
+    /**
+     * @description Checks if the user has access to create or update a Recurring Donation
+     * @return Boolean
+     */
+    private static Boolean hasAccess() {
+        String rdObjectName = Schema.SObjectType.npe03__Recurring_Donation__c.getName();
+
+        return UTIL_Permissions.canRead(rdObjectName, false) &&
+            UTIL_Permissions.canCreate(rdObjectName, false) &&
+            UTIL_Permissions.canUpdate(rdObjectName, false);
+    }
+
+    /**
+     * @description Determines if the Recurring Donation is closed
+     * @param rdId Recurring Donation Id
+     * @return Boolean
+     */
+    private static Boolean isClosed(Id rdId) {
+        Boolean isClosed = false;
+        try {
+            npe03__Recurring_Donation__c rd = [
+                SELECT Status__c
+                FROM npe03__Recurring_Donation__c
+                WHERE Id = :rdId
+                LIMIT 1
+            ];
+
+            isClosed = (new RD2_RecurringDonation(rd)).isClosed();
+
+        } catch (Exception e) {
+            //the record has been deleted; 
```

Comment:
Though unlikely, is it worth throwing an AuraHandledException here, even if it just returns `e.getMessages()[0]`?